spec/services/course/survey/survey_download_spec.rb

```diff
@@ -0,0 +1,224 @@
+        ]
+      end
+
+      let!(:answers) do
+        [
+          create(:course_survey_answer, question: questions[0], response: response, text_response: 'Q1 Answer'),
+          create(:course_survey_answer, question: questions[1], response: response, text_response: 'Q2 Answer'),
+          create(:course_survey_answer, question: questions[2], response: response, text_response: 'Q3 Answer')
+        ]
+      end
+
+      it 'returns an array with four more elements than the no. of questions' do
+        expect(subject.size).to eq(questions.size + 4)
+      end
+
+      it 'first four elements are timestamp, course user id, name, and role' do
+        expect(subject.slice(0..3)).to eq(
+          [
+            response.submitted_at,
+            response.course_user.id,
+            response.course_user.name,
+            response.course_user.role
+          ]
+        )
+      end
+
+      it 'returns answers that correspond to questions' do
+        expect(subject.slice(4..-1)).to eq(['Q1 Answer', 'Q2 Answer', 'Q3 Answer'])
+      end
+    end
+
+    describe '#generate_value' do
+      subject do
+        Course::Survey::SurveyDownloadService.send(:generate_value, answer)
+      end
+
+      let(:response) do
+        create(:course_survey_response, survey: survey, creator: student,
+                                        submitted_at: Time.zone.now)
+      end
+
+      context 'MRQ question' do
+        let(:question) do
+          section = create(:course_survey_section, survey: survey)
+          create(:course_survey_question, question_type: :multiple_response, section: section)
+        end
+        let(:options) do
+          [
+            create(:course_survey_question_option, question: question, option: 'Cool', weight: 5),
+            create(:course_survey_question_option, question: question, option: 'Nice', weight: 2),
+            create(:course_survey_question_option, question: question, option: 'Ok', weight: 3)
+          ]
+        end
+        let(:answer) do
+          create(:course_survey_answer, question: question, response: response)
+        end
+
+        context 'all options are selected' do
+          before do
+            options.each do |question_option|
+              answer.options.build(question_option: question_option)
+            end
+          end
+
+          it 'joins all options with semicolon in increasing weight' do
+            expect(subject).to eq('Nice;Ok;Cool')
+          end
+        end
+
+        context 'no options are selected' do
+          before do
+            answer.options.destroy_all
+          end
+
+          it 'returns an empty string' do
+            expect(subject).to eq('')
+          end
+        end
+      end
+
+      context 'MCQ question' do
+        let(:question) do
+          section = create(:course_survey_section, survey: survey)
+          create(:course_survey_question, question_type: :multiple_choice, section: section)
+        end
+        let(:options) do
```

Comment:
This can be shifted out to the `describe` block, so only 1 set of options needs to be defined for both the MRQ and MCQ contexts instead of duplicating the same declaration.

I think it'll take the `question` that's defined in each context. Try it and see.